tests/test_checkout.py

```diff
@@ -8,64 +8,207 @@ class TestCheckout(object):
     def test_two_write_checkouts(self, repo):
         w1_checkout = repo.checkout(write=True)
         with pytest.raises(PermissionError):
-            w2_checkout = repo.checkout(write=True)
+            repo.checkout(write=True)
         w1_checkout.close()
 
     def test_two_read_checkouts(self, repo, array5by7):
         w_checkout = repo.checkout(write=True)
         dataset_name = 'dset'
         r_ds = w_checkout.datasets.init_dataset(name=dataset_name, prototype=array5by7)
         r_ds['1'] = array5by7
+        w_checkout.metadata.add('init', 'array5by7 added')
         w_checkout.commit('init')
         r1_checkout = repo.checkout()
         r2_checkout = repo.checkout()
         assert np.allclose(r1_checkout.datasets['dset']['1'], array5by7)
-        assert np.allclose(r1_checkout.datasets['dset']['1'], r2_checkout.datasets['dset']['1'])
+        assert np.allclose(
+            r1_checkout.datasets['dset']['1'], r2_checkout.datasets['dset']['1'])
+        assert r1_checkout.metadata.get('init') == 'array5by7 added'
+        assert r2_checkout.metadata.get('init') == 'array5by7 added'
         r1_checkout.close()
         r2_checkout.close()
         w_checkout.close()
 
     def test_write_with_r_checkout(self, written_repo, array5by7):
```

Comment:
style note: no abbreviations in test case function names. 

NOTE: this applies for all test functions, no need for me to clutter the review with comments for every single one.